src/simulator/views/map/display/online_lstm_map_display.py

```diff
@@ -57,32 +53,20 @@ def render(self, screen: pygame.Surface) -> bool:
     tensor(-1.3146)
     """
 
-    def __render_arc(self, p1: Point, p2: Point, screen: pygame.Surface,
-                      color: Union[pygame.Color, Tuple[int, int, int]]):
-        point = self.get_renderer_view().get_screen_rect(p1).center
+    def __render_arc(self, p1: Point, p2: Point, colour: Colour):
+        import pygame  # todo: port this to Panda3D
+
+        point = self.get_renderer_view().get_center(p1)
         size = 100
         area = pygame.Rect(point[0] - size / 2, point[1] - size / 2, size, size)
         dir = p2.to_tensor() - p1.to_tensor()
-        angle = torch.atan2(-dir[1], dir[0]) # pygame
+        angle = torch.atan2(-dir[1], dir[0])  # pygame
         angle %= 2 * np.pi
-        # print(p1, angle)
-        self.services.render_engine.draw_arc(screen, color, area, 0, angle, 2)
-
-    def __render_vector(self, origin: Point, magnitude: float, angle: float, screen: pygame.Surface,
-                        color: Union[pygame.Color, Tuple[int, int, int]]):
-        rot_mat = np.array([
-            [np.cos(angle), -np.sin(angle)],
-            [np.sin(angle), np.cos(angle)],
-        ])
-        unit_vector = np.array([1, 0]) * magnitude
-        vec = origin + np.matmul(rot_mat, unit_vector)
-        self.__render_line(origin, Point(*vec), screen, color)
-
-    def __render_line(self, p1: Point, p2: Point, screen: pygame.Surface,
-                      color: Union[pygame.Color, Tuple[int, int, int]]):
-        first_point = self.get_renderer_view().get_screen_rect(p1).center
-        second_point = self.get_renderer_view().get_screen_rect(p2).center
-        self.services.render_engine.draw_line(screen, color, first_point, second_point, 2)
+
+        self.services.render_engine.draw_arc(color, area, 0, angle, 2)
```

Comment:
Rather than using pygame for this, can we just raise a NotImplementedError or something similar until we get a draw_arc method for the panda3d backend?

Reply:
Changed.